Sources/Contour/Filters/MessageExchangeFilterInvoker.cs

```diff
@@ -25,15 +25,15 @@ public class MessageExchangeFilterInvoker
         /// The filters.
         /// </param>
         public MessageExchangeFilterInvoker(IEnumerable<IMessageExchangeFilter> filters)
+            : this(filters, null)
         {
-            this.filterEnumerator = filters.Reverse().
-                GetEnumerator();
         }
 
         public MessageExchangeFilterInvoker(IEnumerable<IMessageExchangeFilter> filters, IDictionary<Type, IMessageExchangeFilterDecorator> filterDecorators)
-            : this(filters)
         {
-            this.filterDecorators = filterDecorators;
+            this.filterEnumerator = filters.Reverse().
```

Comment:
The IEnumerable collection does not guarantee the order of its elements. What should the correct order be here?